src/emulate.c

```diff
@@ -76,49 +89,84 @@ static void rv_exception_default_handler(riscv_t *rv)
+            mode = rv->csr_stvec & 0x3;                                       \
+            rv->csr_sepc = rv->PC;                                            \
+            rv->csr_stval = mtval;                                            \
+            rv->csr_scause = code;                                            \
+            rv->csr_sstatus |= SSTATUS_SPP; /* set privilege mode */          \
+        } else { /* machine */                                                \
+            const uint32_t mstatus_mie =                                      \
+                (rv->csr_mstatus & MSTATUS_MIE) >> MSTATUS_MIE_SHIFT;         \
+            rv->csr_mstatus |= (mstatus_mie << MSTATUS_MPIE_SHIFT);           \
+            rv->csr_mstatus &= ~(MSTATUS_MIE);                                \
+            rv->csr_mstatus |= (rv->priv_mode << MSTATUS_MPP_SHIFT);          \
+            rv->priv_mode = RV_PRIV_M_MODE;                                   \
+            base = rv->csr_mtvec & ~0x3;                                      \
+            mode = rv->csr_mtvec & 0x3;                                       \
+            rv->csr_mepc = rv->PC;                                            \
+            rv->csr_mtval = mtval;                                            \
+            rv->csr_mcause = code;                                            \
+            rv->csr_mstatus |= MSTATUS_MPP; /* set privilege mode */          \
+            if (!rv->csr_mtvec) { /* in case CSR is not configured */         \
+                rv_interrupt_exception_default_handler(rv);                   \
+                return;                                                       \
+            }                                                                 \
+        }                                                                     \
+        switch (mode) {                                                       \
+        /* DIRECT: All exceptions set PC to base */                           \
+        case 0:                                                               \
+            rv->PC = base;                                                    \
+            break;                                                            \
+        /* VECTORED: Asynchronous interrupts set PC to base + 4 * code */     \
+        case 1:                                                               \
+            /* MSB of code is used to indicate whether the trap is interrupt  \
+             * or exception, so it is not considered as the 'real' code */    \
+            rv->PC = base + 4 * (code & MASK(31));                            \
+            break;                                                            \
+        }                                                                     \
+	/* block escaping for interrupt or exception handling */ \
+        if(rv->is_interrupted){ \
+	    assert(setjmp(env) == 0); \
```

Comment:
If `NDEBUG` is set, `setjmp` will not be called.

Reply:
I'll make it into an if conditional statement.